testcases/Crashtool.py

```diff
@@ -0,0 +1,131 @@
+        self.commands = ["log", "bt", "ps", "runq", "kmem -i", "kmem -o", "kmem -h", "vm"]
+
+    def run_crash_command(self, cmd, timeout=180):
+        """
+        1.Sends a command to the running crash session
+        2.Waits for the crash> prompt again
+        3.Captures and returns command output, stripping echo
+        4.If timeout or error occurs, raises an exception unless the prompt is found
+        """
+        self.console.sendline(cmd)
+        try:
+            self.console.expect_exact("crash>", timeout=180)
+            full_output = self.console.before.strip()
+            lines = full_output.splitlines()
+            if lines and lines[0].strip() == cmd:
+                lines = lines[1:]
+            output = "\n".join(lines).strip()
+            return output
+        except Exception as e:
+            if "crash>" in self.console.before:
+                full_output = self.console.before.strip()
+                lines = full_output.splitlines()
+                if lines and lines[0].strip() == cmd:
+                    lines = lines[1:]
+                return "\n".join(lines).strip()
+            else:
+                raise Exception(f"Crash command '{cmd}' timed out or failed") from e
+
+    def runTest(self):
+        """
+        1.Gets kernel version (uname -r)
+        2.Finds latest crash directory (e.g., /var/crash/2025-06-03-07-08/)
+        3.Validates vmcore file type and size
+        4.Detects OS type/version and constructs appropriate vmlinux path
+        5.Launches crash interactively
+        6.Runs commands (stored in self.commands)
+        7.Exits crash cleanly
+        """
+        kernel_version = self.cv_HOST.host_run_command("uname -r")[0].strip()
+        crash_dir = self.cv_HOST.host_run_command("basename $(ls -1td /var/crash/*/ | head -n 1)")[0].strip()
+        vmcore_path = f"/var/crash/{crash_dir}/vmcore"
+        file_info = self.cv_HOST.host_run_command(f"file {vmcore_path}")
+        size_check = self.cv_HOST.host_run_command(f"stat -c %s {vmcore_path}")
+        vmcore_size = int(size_check[0].strip())
+
+        if "vmcore" not in file_info[0] or vmcore_size < 100000:
+            self.fail("Invalid or corrupted vmcore")
+        res = self.cv_HOST.host_run_command("cat /etc/os-release", timeout=60)
+        version = self.cv_HOST.host_run_command("awk -F '\"' '/^VERSION_ID=/ { print $2 }' /etc/os-release | cut -d. -f1")[0].strip()
+        if 'Red Hat' in res[0] or 'Red Hat' in res[1]:
+            vmlinux = f"/usr/lib/debug/lib/modules/{kernel_version}/vmlinux"
+        elif ('SLES' in res[0] or 'SLES' in res[1]) and version == "16":
+            vmlinux = f"/boot/vmlinux-{kernel_version}"
+            debug_info = f"/usr/lib/debug/usr/lib/modules/{kernel_version}/vmlinux.debug"
+            vmlinux = f"{vmlinux} {debug_info}"
+
+        elif 'SLES' in res[0] or 'SLES' in res[1]:
+            vmlinux = f"/boot/vmlinux-{kernel_version}"
```

Comment:
Don't we need debug vmlinux with previous SLES releases?

Reply:
tested on sles15sp7 .. it worked fine